go/processor/geth/state_contract.go

```diff
@@ -0,0 +1,216 @@
+)
+
+var (
+	setBalanceMethodID []byte
+	copyCodeMethodID   []byte
+	swapCodeMethodID   []byte
+	setStorageMethodID []byte
+	incNonceMethodID   []byte
+)
+
+func init() {
+	abi, err := abi.JSON(strings.NewReader(contractABI))
+	if err != nil {
+		panic(err)
+	}
+
+	for name, constID := range map[string]*[]byte{
+		"setBalance": &setBalanceMethodID,
+		"copyCode":   &copyCodeMethodID,
+		"swapCode":   &swapCodeMethodID,
+		"setStorage": &setStorageMethodID,
+		"incNonce":   &incNonceMethodID,
+	} {
+		method, exist := abi.Methods[name]
+		if !exist {
+			panic("unknown EvmWriter method")
+		}
+
+		*constID = make([]byte, len(method.ID))
+		copy(*constID, method.ID)
+	}
+}
+
+type PreCompiledContract struct{}
+
+func (PreCompiledContract) Run(stateDB vm.StateDB, _ vm.BlockContext, txCtx vm.TxContext, caller common.Address, input []byte, suppliedGas uint64) ([]byte, uint64, error) {
```

Comment:
nit: there is repeated code in the different branches of the if that could be abstracted.